src/registry.cpp

```diff
@@ -251,38 +284,52 @@ void jitc_registry_set_attr(void *ptr, const char *name,
         size_t new_size = (size_t) new_count * (size_t) isize;
         void *ptr;
 
-        if (!state.devices.empty()) {
+        if (backend == JitBackend::CUDA) {
             scoped_set_context guard(state.devices[0].context);
-            CUresult ret = cuMemAllocManaged((CUdeviceptr *) &ptr, new_size, CU_MEM_ATTACH_GLOBAL);
+            CUresult ret = cuMemAlloc((CUdeviceptr *) &ptr, new_size);
             if (ret != CUDA_SUCCESS) {
                 jitc_malloc_trim(true, true);
-                cuda_check(cuMemAllocManaged((CUdeviceptr *) &ptr, new_size, CU_MEM_ATTACH_GLOBAL));
+                cuda_check(cuMemAlloc((CUdeviceptr *) &ptr, new_size));
             }
-            cuda_check(cuMemAdvise((CUdeviceptr) ptr, new_size, CU_MEM_ADVISE_SET_READ_MOSTLY, 0));
+
+            // cuda_check(cuCtxSynchronize());
```

Comment:
This can be removed now?

Reply:
Haven't tried this on Windows yet. But it should be yes